airflow/models/taskinstance.py

```diff
@@ -122,6 +122,9 @@ def clear_task_instances(tis,
             dr.start_date = timezone.utcnow()
 
 
+TaskInstanceKey = Tuple[str, str, datetime, int]
```

Comment:
Yes to this change, but can you also use it in the executors/task runners?

Oh can you also add a (doc?) comment saying what the str,str and int are?

Reply:
I will add documentation, but in other places I would prefer not to fix it. Currently Jarek deals with this and checking each place thoroughly is very difficult.

Reply:
We want to introduce named tuple, but first we have to solve problems with cyclical imports.

Reply:
Yeah. I will do it soon/